cocos/renderer/CCRenderer.cpp

```diff
@@ -845,11 +870,18 @@ void Renderer::drawBatchedTriangles()
     /************** 3: Draw *************/
     for (int i=0; i<batchesTotal; ++i)
     {
-        CC_ASSERT(_triBatchesToDraw[i].cmd && "Invalid batch");
-        _triBatchesToDraw[i].cmd->useMaterial();
-        glDrawElements(GL_TRIANGLES, (GLsizei) _triBatchesToDraw[i].indicesToDraw, GL_UNSIGNED_SHORT, (GLvoid*) (_triBatchesToDraw[i].offset*sizeof(_indices[0])) );
-        _drawnBatches++;
-        _drawnVertices += _triBatchesToDraw[i].indicesToDraw;
+        //CC_ASSERT(_triBatchesToDraw[i].cmd && "Invalid batch");
```

Comment:
In this case, the use `CC_ASSERT()` is more right (even if it's involves the signal, so this is right behaviour for `assert()` ) than through the use of if..else constructions, since CC_ASSERT identifies the existence of an architectural issue, and second condition only hide its presence.